src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -318,6 +324,79 @@ def fuzz(self,
     return LibFuzzerCommon.fuzz(self, corpus_directories, fuzz_timeout,
                                 artifact_prefix, additional_args)
 
+class FuchsiaQemuLibFuzzerRunner(new_process.ProcessRunner,LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform)."""
+  def __init__(self, executable_path, default_args=None):
+    # This code assumes the following layout for files in the storage bucket:
+    # * multiboot.bin
+    # * fvm.blk
+    # * fuchsia-ssh.zbi
+    # * qemu/*
+    # * .ssh/*
+    local_path = os.getcwd() + "/"
+    resources_path = local_path + "fuchsia_on_clusterfuzz_resources_v1"
+
+    try:
+      os.mkdir(resources_path)
+    except OSError as e:
+      if e.errno == errno.EEXIST:
+        pass
+    subbed_fuchsia_gsutil_command = [param.replace("{fuchsia_resources_path}", environment.get_value('FUCHSIA_RESOURCES_PATH', ''))
+                                     .replace("{local_resources_path}", resources_path) for param in constants.FUCHSIA_GSUTIL_COMMAND]
+    subprocess.call(subbed_fuchsia_gsutil_command)
+    qemu_path = resources_path + "/qemu-for-fuchsia/bin/qemu-system-x86_64"
+    os.chmod(qemu_path, 0o550)
+    kernel_path = resources_path + "/multiboot.bin"
+    os.chmod(kernel_path, 0o644)
+    pkey_path = resources_path + "/.ssh/pkey"
+    os.chmod(pkey_path, 0o400)
+    sharefiles_path = resources_path + "/qemu-for-fuchsia/share/qemu"
+    initrd_path = resources_path + "/fuchsia-ssh.zbi"
+    os.chmod(initrd_path, 0o644)
+    drive_path = resources_path + "/fuchsia.qcow2"
+    os.chmod(drive_path, 0o644)
+
+    # TODO: Add a mechanism for choosing portnum dynamically.
+    portnum = "56339"
+
+    self.subbed_qemu_base_command = [param.replace("{qemu}", qemu_path)
+                                .replace("{kernel}", kernel_path)
+                                .replace("{drive}", drive_path)
+                                .replace("{initrd}", initrd_path)
+                                .replace("{sharefiles}", sharefiles_path)
+                                .replace("{portnum}", portnum) for param in constants.FUCHSIA_QEMU_COMMAND_TEMPLATE]
+    self.subbed_ssh_base_command = [param.replace("{identity_file}", pkey_path)
```

Comment:
See my comment on line 362 about using `.format` instead of this technique.
If you need a list you can call `s.split(' ')` to separate the string into a list by each space.

Reply:
see discussion on L362 comments